clang/lib/CIR/CodeGen/CIRGenExprScalar.cpp

```diff
@@ -148,3 +150,27 @@ mlir::Value ScalarExprEmitter::VisitCastExpr(CastExpr *ce) {
   }
   return {};
 }
+
+/// Return the size or alignment of the type of argument of the sizeof
+/// expression as an integer.
+mlir::Value ScalarExprEmitter::VisitUnaryExprOrTypeTraitExpr(
+    const UnaryExprOrTypeTraitExpr *e) {
+  const QualType typeToSize = e->getTypeOfArgument();
+  if (e->getKind() == UETT_SizeOf) {
```

Comment:
See here: https://github.com/llvm/llvm-project/blob/dc23234a6613d304014f7bfc9f36b55739401270/clang/lib/CodeGen/CGExprScalar.cpp#L3471

This is missing A LOT of those things it needs.  I think this function needs more work (even if it is just NYI).